Assignment 2/src/main/java/com/nuclei/assignment/service/diskstorage/DiskStorageOperationImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.nuclei.assignment.service.diskstorage;
+
+import com.nuclei.assignment.constants.ExceptionsConstantsUtils;
+import com.nuclei.assignment.constants.StringConstantsUtils;
+import com.nuclei.assignment.constants.SuccessConstantsUtils;
+import com.nuclei.assignment.entity.UserEntity;
+import com.nuclei.assignment.exception.CustomException;
+
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Disk Storage Operation implementation.
+ */
+public class DiskStorageOperationImpl implements DiskStorageOperation {
+  
+  @Override
+  public void saveUsersToDisk(final List<UserEntity> users) throws CustomException {
+    try (OutputStream file =
+           Files.newOutputStream(Paths.get(StringConstantsUtils.USER_STORAGE))) {
+      try (ObjectOutputStream writer = new ObjectOutputStream(file)) {
+        writer.writeObject(users);
+      } catch (Exception exception) {
+        throw new CustomException(ExceptionsConstantsUtils.FAILED_TO_WRITE_STORAGE,
+          exception);
+      }
+    } catch (Exception exception) {
+      throw new CustomException(ExceptionsConstantsUtils.FAILED_TO_WRITE_STORAGE,
+          exception);
+    }
+  }
+  
+  @Override
+  @SuppressWarnings("unchecked")
```

Comment:
Don't suppress, fix the reason that causing this.

Reply:
done